### PR TITLE
Content refactor

### PR DESCRIPTION
**Pull request by rtibbles**
### Summary
* Refactors content models to clearly disambiguate imported content metadata from locally generated metadata. Makes no actual change to the schema, just how it is represented in code.
* Fixes a bug in the Postgresql backed queue implementation whereby initializing the DB connection would fail when the port was left undefined (and therefore meant to be the postgres default value).
* Adds the capability to set a target for content import (aside from the default DB), which would allow for schema conversion of imported channel DBs to allow for non-destructive previewing of updated channel DBs.
* Fixes generation of import schema through the `generate_schema` management command, so that we can once again make updates to the import content schema.

### Reviewer guidance
Most importantly is the tests all passing.
Secondly is if the doc strings in the new model modules are sufficient.

### References
No direct references.

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [x] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by rtibbles** [comment]
So glad I have good test coverage on this - will fix up the test issues.

**Comment by rtibbles** [comment]
Have isolated the test error here to postgres sessions not getting disconnected, not with the primary behaviour. Will continue to investigate.

**Comment by rtibbles** [comment]
I have updated this PR to separate out the Postgres speed up work which seems to be hitting some test failures (only on Travis, I fixed all the ones I could replicate locally), so that the work to support changes to the import export workflow can be merged sooner.

File: kolibri/core/content/utils/channel_import.py [hunk 1]
```diff
@@ -115,9 +117,19 @@ def __init__(self, channel_id, channel_version=None, cancel_check=None):
...
-            schema_version=CONTENT_SCHEMA_VERSION, app_name=CONTENT_APP_NAME
-        )
+        if destination is None:
+            # If no destination is set then we are targeting the default database
+            self.destination = Bridge(
+                schema_version=CONTENT_SCHEMA_VERSION, app_name=CONTENT_APP_NAME
+            )
+        else:
+            # If a destination is set then pass that explicitly. At the moment, this only supports
+            # importing to an arbitrary SQLite file path.
```
Comment (ralphiee22) [inline kolibri/core/content/utils/channel_import.py:127]: Assuming we don't need any SQLite specific code checks?

**Review by ralphiee22 — APPROVED** [review]
Haven't tested locally, but code looks good.

File: kolibri/core/content/utils/channel_import.py [hunk 2]
```diff
@@ -469,7 +469,12 @@ def check_and_delete_existing_channel(self):
                         new_channel_version=self.channel_version,
                     )
                 )
-                self.delete_old_channel_data(existing_channel.root.tree_id)
+
+                root_node = self.destination.session.query(
```
Comment (ralphiee22) [inline kolibri/core/content/utils/channel_import.py:473]: is a regression test needed for this?

File: kolibri/core/content/utils/channel_import.py (hunk 1 above)
Comment (rtibbles) [reply kolibri/core/content/utils/channel_import.py:127]: We might want to add some extra guards to ensure that this is only a SQLite file path, but this is more of a proof of concept at this point, as there is no practical way to access this code path.

File: kolibri/core/content/utils/channel_import.py (hunk 2 above)
Comment (rtibbles) [reply kolibri/core/content/utils/channel_import.py:473]: Possibly, I've opened a follow up issue here: https://github.com/learningequality/kolibri/issues/5960

**Pull request merged**